packages/track-event-mixin/track-event-gesture.ts

```diff
@@ -0,0 +1,308 @@
+// Check for touch-only devices
+const IS_TOUCH_ONLY = navigator.userAgent.match(/iP(?:[oa]d|hone)|Android/);
+// Number of last N track positions to keep
+const TRACK_LENGTH = 2;
+// Radius for track
+const TRACK_DISTANCE = 5;
+// Stored touch identifier
+const TOUCH_STATE = {
+  id: -1
+};
+
+type TrackMove = {
+  x: number;
+  y: number;
+};
+
+type TrackGestureInfo = {
+  x: number;
+  y: number;
+  state: 'start' | 'track' | 'pan-x' | 'pan-y' | 'end';
+  started: boolean;
+  moves: TrackMove[];
+  addMove: (move: TrackMove) => void;
+  moveFn?: (event: MouseEvent) => void;
+  upFn?: (event: MouseEvent) => void;
+};
+
+type TrackGesture = {
+  name: string;
+  touchAction: string;
+  deps: Array<'mousedown' | 'touchstart' | 'touchmove' | 'touchend'>;
+  info: TrackGestureInfo;
+  reset: () => void;
+  mousedown: (event: MouseEvent) => void;
+  touchstart: (event: TouchEvent) => void;
+  touchmove: (event: TouchEvent) => void;
+  touchend: (event: TouchEvent) => void;
+};
+
+function isMouseEvent(name: string): boolean {
+  return ['mousedown', 'mousemove', 'mouseup'].indexOf(name) > -1;
```

Comment:
I'd prefer this since we don't need to support IE11 here anymore:
```suggestion
  return ['mousedown', 'mousemove', 'mouseup'].includes(name);
```

Reply:
Done.